src/librustc/middle/const_eval.rs

```diff
@@ -416,6 +424,12 @@ pub enum ErrKind {
     ExpectedConstTuple,
     ExpectedConstStruct,
     TupleIndexOutOfBounds,
+    IndexedNonVec,
+    IndexNotNatural,
```

Comment:
(though perhaps this error is also issues when the index is a floating point number, i.e. a non-integer?  That's the other reason one would say "Natural" ...)